### PR TITLE
filter (ticdc): fix synpoint table is replicated in BDR mode

### LINKED ISSUE #10576: TiCDC: The sync-point table got replicated (esp. in bdr-mode)
### What did you do?

1. Create two clusters

    ```sh
    tiup playground v7.6.0 --host 10.0.0.2 --db 1 --kv 1 --pd 1 --ticdc 1 --tiflash 0 --without-monitor
    rm /tmp/tidb-4000.sock
    tiup playground v7.6.0 --host 10.0.0.48 --db 1 --kv 1 --pd 1 --ticdc 1 --tiflash 0 --without-monitor
    ```

2. Setup two changefeeds between them

    ```toml
    # 2to48.toml
    bdr-mode = true
    enable-sync-point = true
    sync-point-interval = "31s"
    ```

    ```toml
    # 48to2.toml
    bdr-mode = true
    enable-sync-point = true
    sync-point-interval = "37s"
    ```

    ```sh
    tiup cdc:v7.6.0 cli changefeed create --pd http://10.0.0.2:2379 --sink-uri mysql://root@10.0.0.48 -c 2to48 --config 2to48.toml
    tiup cdc:v7.6.0 cli changefeed create --pd http://10.0.0.48:2379 --sink-uri mysql://root@10.0.0.2 -c 48to2 --config 48to2.toml
    ```

3. Wait for several minutes (because `minSyncPointInterval` is 30s 🤷)

4. Check the syncpoint tables

    ```sh
    mysql --host 10.0.0.2 --port 4000 -u root -e 'select * from tidb_cdc.syncpoint_v1;'
    mysql --host 10.0.0.2 --port 4000 -u root -e 'select count(*) from tidb_cdc.syncpoint_v1;'

    mysql --host 10.0.0.48 --port 4000 -u root -e 'select * from tidb_cdc.syncpoint_v1;'
    mysql --host 10.0.0.48 --port 4000 -u root -e 'select count(*) from tidb_cdc.syncpoint_v1;'
    ```

### What did you expect to see?

The syncpoint table in 10.0.0.2 only contained entries of the 48to2 changefeed, 
and similarly 10.0.0.48 only contained entries of the 2to48 changefeed.

### What did you see instead?

Both clusters contained sync-point entries from both changefeeds.

### Versions of the cluster

v7.6.0

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #10576 

### What is changed and how it works?
Make isSysSchema() also treat `tidb_cdc` as a system schema. 
This also helps the case of non-BDR "A→B→C" scenario, where the syncpoint table of A→B polluted B→C.
When the schema `tidb_cdc` is considered as a system schema, it will not be listened to by CDC at all.

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
Fix a bug that causes syncpoint table being replicated by cdc.
```

### CODE REVIEW

**Review by CharlesCheung96 — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

File: pkg/filter/utils.go [hunk 1]
```diff
@@ -27,7 +27,14 @@ import (
 
 // isSysSchema returns true if the given schema is a system schema
 func isSysSchema(db string) bool {
-	return tifilter.IsSystemSchema(db)
+	switch db {
+	// schema `tidb_cdc` is used in syncpoint function.
+	// Tables in `tidb_cdc` should not be replicated by cdc.
+	case "tidb_cdc":
```
Comment (flowbehappy) [inline pkg/filter/utils.go:33]: Use a constant var please. And also update the code in syncpoint module.

**Comment by flowbehappy** [comment]
I think we need a BDR test with synpoint enable.

File: pkg/filter/utils.go (hunk 1 above)
Comment (asddongmen) [reply pkg/filter/utils.go]: done

**Comment by asddongmen** [comment]
> I think we need a BDR test with synpoint enable.

Added in integration test.

**Review by kennytm — APPROVED** [review]

File: pkg/filter/utils_test.go [hunk 2]
```diff
@@ -42,6 +42,7 @@ func TestIsSchema(t *testing.T) {
 		{tifilter.InspectionSchemaName, true},
 		{tifilter.PerformanceSchemaName, true},
 		{tifilter.MetricSchemaName, true},
+		{"tidb_cdc", true},
```
Comment (kennytm) [inline pkg/filter/utils_test.go:45]: maybe use `TiCDCSystemSchema` instead of `"tidb_cdc"` for this one

Comment (asddongmen) [reply pkg/filter/utils_test.go:45]: Good catch, thank you.

**Pull request merged**